kaldb/src/main/java/com/slack/kaldb/metadata/service/ServiceMetadata.java

```diff
@@ -0,0 +1,92 @@
+package com.slack.kaldb.metadata.service;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ImmutableList;
+import com.slack.kaldb.metadata.core.KaldbMetadata;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Configurations for an upstream service for use in the pre-processor - including rate limits, and
```

Comment:
This comment isn't quite accurate since this data is also used in query service. I think it's better to just add the explanations in protobuf file. 

In this specific file, we can say this is a wrapper class that stores the service specific metadata and the constraints on partition list.

Reply:
Updated comment to reflect this includes the query service as well. I'm not a fan of stating this is a wrapper class for service metadata that contains a partition list, as that just restates what is obvious from the class name and fields.